lib/vanagon/project.rb

```diff
@@ -476,5 +477,36 @@ def generate_package
     def generate_packaging_artifacts(workdir)
       @platform.generate_packaging_artifacts(workdir, @name, binding)
     end
+
+    # Generate a json file which lists all of the dependant components of the project
+    def generate_dependencies_info
+      project_components = @components.map do |component|
+        components_hash = component.options
+        [:version, :url].each do |key|
+          if val = component.send(key)
+            components_hash[key] = val
+          end
+        end
+        [
+          component.name,
+          components_hash,
+        ]
```

Comment:
(specifically, there's no documentation about what this method should return. It looks like you're going to build a multidimensional Array and then cast it to a Hash. You'd get something like this back:

```ruby
>> [["a", {b: nil, c: nil}],["b", {c: nil, d: nil}]].to_h
=> {
  "a" => {
    :b => nil,
    :c => nil
  },
  "b" => {
    :c => nil,
    :d => nil
  }
}
```

That makes sense and I think it's a solid approach. But the path to getting there is kind of winding since you're constructing everything in the Project and offloading nothing to the Components themselves.)